stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -70,6 +70,10 @@ function(_add_target_variant_c_compile_link_flags)
     endif()
   endif()
 
+  if("${CFLAGS_SDK}" STREQUAL "ANDROID")
+    set(DEPLOYMENT_VERSION ${SWIFT_ANDROID_API_LEVEL})
```

Comment:
Setting this variable adds the API level to the target triple just below, which then enables clang to automatically find the right Android libraries when linking.